xsuite/src/proxy/proxy.ts

```diff
@@ -29,22 +29,38 @@ export class Proxy {
     this.blockNonce = params.blockNonce ?? null;
   }
 
-  fetchRaw(path: string, data?: any) {
-    const baseUrl = this.proxyUrl;
-    const init: RequestInit = { headers: this.headers };
-    const url = new URL(`${baseUrl}${path}`);
-    if (this.blockNonce !== null) {
-      url.searchParams.append("blockNonce", this.blockNonce.toString());
+  private makeUrl(
+    path: string | URL,
+    params: Record<string, { toString: () => string } | undefined | null> = {},
```

Comment:
In which case would we want `null` to be allowed?

Reply:
Right now it's useful for the `blockNonce` param, we just give the `blockNonce` to `makeUrl` and it doesnt add it if its `null`